cmd/secrethelper/providers/k8s_secret_test.go

```diff
@@ -74,7 +75,15 @@ func TestReadKubernetesSecret(t *testing.T) {
 			}
 			kubeClient := fake.NewSimpleClientset(kubeObjects...)
 
-			resolvedSecret := ReadKubernetesSecret(kubeClient, test.secretPath)
+			secretGetter := func(namespace, name string) (map[string][]byte, error) {
+				secret, err := kubeClient.CoreV1().Secrets(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+				if err != nil {
+					return nil, err
+				}
+				return secret.Data, nil
+			}
+
+			resolvedSecret := ReadKubernetesSecret(secretGetter, test.secretPath)
```

Comment:
I tried not to change any logic in the tests, they can keep using the k8s client, it won't impact builds